### PR TITLE
extensions: facet

### PR DESCRIPTION
**Pull request by jrcastro2**
* Adds extensions facet
* Allows indexing of extensions

### CODE REVIEW

File: invenio_app_ils/config.py [hunk 1]
```diff
@@ -831,6 +832,9 @@ def _(x):
             ),
             medium=dict(terms=dict(field="stock.mediums")),
         ),
+        filters=dict(
+            extensions=custom_metadata_filter('extensions'),
```
Comment (ntarocco) [inline invenio_app_ils/config.py]: remove?

File: invenio_app_ils/documents/facets.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+"""ILS Document indexer APIs."""
+
+import re
+
+from elasticsearch_dsl import Q
+from flask import current_app
+from invenio_rest.errors import FieldError, RESTValidationError
+
+
+def custom_metadata_filter(field):
```
Comment (ntarocco) [inline invenio_app_ils/documents/facets.py]: is this to be removed after our chat?  
I am sorry if you spent a lot of time on this code :(

File: invenio_app_ils/documents/mappings/v6/documents/document-v1.0.0.json [hunk 3]
```diff
@@ -259,9 +259,7 @@
           "type": "object"
         },
         "extensions": {
-          "type": "object",
-          "dynamic": false,
```
Comment (ntarocco) [inline invenio_app_ils/documents/mappings/v6/documents/document-v1.0.0.json:263]: I would probably have left `dynamic: false`... any particular reason why you have removed it?

**Review by ntarocco — COMMENTED** [review]
Did you check if there is nothing to change in the React app when rendering the custom fields, both in the details page (frontsite and backend) and also in the edit/create form?
The `:` might be hardcoded somewhere...

File: invenio_app_ils/documents/mappings/v6/documents/document-v1.0.0.json (hunk 3 above)
Comment (jrcastro2) [reply invenio_app_ils/documents/mappings/v6/documents/document-v1.0.0.json:263]: If we keep `dynamic: false` we cannot search for the extensions in the search-box. I didn't check further.

**Comment by jrcastro2** [comment]
> Did you check if there is nothing to change in the React app when rendering the custom fields, both in the details page (frontsite and backend) and also in the edit/create form?
> The `:` might be hardcoded somewhere...

I just double checked and it is working correctly both, frontsite and backoffice, but I noticed that there are 2 fields, [`institution`](https://github.com/CERNDocumentServer/cds-ils/blob/master/cds_ils/config.py#L439-L441) and [`study`](https://github.com/CERNDocumentServer/cds-ils/blob/master/cds_ils/config.py#L447-L450) that are displayed as text but expect a list. (This values are not even migrated as they do not exist in legacy data if I am not mistake. Also this "bug" is not motivated by this change).

image

**Review by ntarocco — APPROVED** [review]

**Pull request merged**